### PR TITLE
Deploy and test scripts MVP

### PR DESCRIPTION
**Pull request by mmichael**

### CODE REVIEW

**Comment by mmichael** [comment]
We decided to move away overlays flushonce and prometheus and use seds to fill in the collector template. Our reason was there wasn't a straight forward transform/merge of the collector.yaml file.

File: hack/kustomize/test.sh [hunk 1]
```diff
@@ -0,0 +1,84 @@
+#! /bin/bash
+
+# This script automates the functional testing of the collector
+
+DEFAULT_VERSION="1.2.7-beta1"
```
Comment (vikramraman) [inline hack/kustomize/test.sh:5]: Can we make these as arguments to this script (It already accepts cluster and token, and will also need these).

File: hack/kustomize/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+# This script automates the deployment of the collector to a specific k8s cluster
+
+DEFAULT_IMAGE_NAME="wavefronthq\/wavefront-kubernetes-collector"
+DEFAULT_VERSION="1.2.6"
+DEFAULT_USE_CLASSIC_PROMETHEUS=false
+DEFAULT_FLUSH_ONCE=false
+DEFAULT_FLUSH_INTERVAL=30
+DEFAULT_COLLECTION_INTERVAL=60
+
+if [[ -z ${FLUSH_ONCE} ]] ; then
```
Comment (vikramraman) [inline hack/kustomize/deploy.sh:12]: So these are just environment vars that one can set?

Comment (mmichael) [reply hack/kustomize/deploy.sh:12]: For now, we wanted to do the simplest thing that worked until it got too painful. Possible next steps once it gets to painful:
1. Use a overrides yaml file and  write a bash or go program to merge an overrides yaml file into collector yaml file
2. Look into other ways would could get the kustomize ConfigMapGenerator to work correctly or see if we can get kustomize patches to update the override values

File: hack/kustomize/test.sh (hunk 1 above)
Comment (mmichael) [reply hack/kustomize/test.sh:5]: Makes sense, we will add them

File: hack/kustomize/overlays/test/collector.yaml.template [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+enableEvents: false
+enableDiscovery: true
+useClassicPrometheusParser: USE_CLASSIC_PROMETHEUS
+flushOnce: FLUSH_ONCE
+flushInterval: FLUSH_INTERVALs
+defaultCollectionInterval: COLLECTION_INTERVALs
+
+
+sinks:
+- proxyAddress: wavefront-proxy.NAMESPACE.svc.cluster.local:2878
```
Comment (vikramraman) [inline hack/kustomize/overlays/test/collector.yaml.template:11]: redirectToLog needs to be templatized as well. It is needed as part of flushOnce.

**Review by vikramraman — APPROVED** [review]

**Pull request merged**